src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -656,10 +776,16 @@ private static void DispatchEx(scoped ref StackFrameIterator frameIter, ref ExIn
                         catchingTryRegionIdx = MaxTryRegionIdx;
```

Comment:
This if statement can be removed, since `handlingFrameSP` and `catchingTryRegionIdx` are being set to the same values below.

```c#
if (pReversePInvokePropagationCallback != IntPtr.Zero)
{
    // Tell the second pass to unwind to this frame.
    handlingFrameSP = frameIter.SP;
    catchingTryRegionIdx = MaxTryRegionIdx;
}
```

Reply:
This is actually a bug I've introduced during final cleanup. They should be set below only for the non-NATIVEAOT case.